### PR TITLE
create user redux store

### PR DESCRIPTION
**Pull request by seanheinrichs**
What's in this PR:

Front End
- Setting up the general store to handle Ajax requets
- Create user actions (GET only)
- Create user reducer
- Create user epics to fire ajax calls
- Create user selectors (isLoading and userDetails)
- Testing in Login Page (will be removed in next PR)

Back End
- Add CORS module

Still TO DO: 
- Create POST Actions to create new User
- Create fake login data to transport user to dashboard where they will see they user information
- Make it look nice

### CODE REVIEW

File: frontend/src/state/user/userEpics.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    getUserError,
+} from './userActions';
+
+// TODO: Hit actual endpoint for users
+
+const getUserEvent = (action$) => {
+	return action$.pipe(
+		ofType(GET_USER_REQUEST),
+		mergeMap(({ payload: { userId } }) =>
+			ajax.getJSON(`http://localhost:3100/users/${userId}/`).pipe(
```
Comment (seanheinrichs) [inline frontend/src/state/user/userEpics.js:22]: will fix this in next PR (add env variable for apiUrl)

File: frontend/src/views/Login.js [hunk 2]
```diff
@@ -45,6 +50,16 @@ const useStyles = makeStyles((theme) => ({
...
+  const dispatch = useDispatch();
+
+  const user = useSelector(userDetailsSelector);
+
+  useEffect(() => {
+    dispatch(getUserRequest(1));
+  }, [])
+
+  // Just Testing To Ensure it works
+  console.log(user);
```
Comment (seanheinrichs) [inline frontend/src/views/Login.js:62]: jsut testing to show that it works haha, will be removed when I make the dashboard

File: frontend/src/state/rootEpic.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { combineEpics } from 'redux-observable';
+
+import userEpics from './user/userEpics';
+
+// Combine all epics into one; this is the expected behaviour of redux-observable.
+const rootEpic = combineEpics(
+    userEpics,
```
Comment (mdesson) [inline frontend/src/state/rootEpic.js:7]: Not a requested change, just a suggestion. Should we add global error handling to catch anything that manages to fall through the other epics uncaught? 

Example near the bottom of the page here:

https://redux-observable.js.org/docs/basics/SettingUpTheMiddleware.html

File: frontend/src/state/user/userEpics.js (hunk 1 above)
Comment (mdesson) [reply frontend/src/state/user/userEpics.js:22]: Sounds good!

File: frontend/src/state/user/userReducer.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+				...state,
+				isLoading: false,
+			};
+
+		default:
+			return state;
+	}
+};
+
+export default userReducer;
```
Comment (mdesson) [inline frontend/src/state/user/userReducer.js:37]: Super clean reducer.

**Pull request merged**